pkgs/battlenet/default.nix

```diff
@@ -0,0 +1,98 @@
+{
+  lib,
+  makeDesktopItem,
+  symlinkJoin,
+  writeShellScriptBin,
+  gamemode,
+  winetricks,
+  wine,
+  dxvk,
+  wineFlags ? "",
+  pname ? "battlenet",
+  location ? "$HOME/Games/battlenet",
+  tricks ? ["arial" "tahoma" "win10"],
+  preCommands ? "",
+  postCommands ? "",
+  pkgs,
+}: let
+  src = pkgs.fetchurl rec {
+    url = "https://www.battle.net/download/getInstallerForGame?os=win&version=LIVE&gameProgram=BATTLENET_APP";
```

Comment:
still figuring out how to get this reproducible ... havent found a versioned link

Reply:
@fufexan do you know if there is a recommended or common way to approach this?

Reply:
I don't. I've looked in the lutris Battle.net script and it's the same URL. If it's not deterministic on their end, there's not much we can do. Maybe there are some other parameters that we don't know about.

Reply:
I just searched for the versions and this URL seems to yield deterministic results, but we'll have to update it periodically in some way.
https://downloader.battle.net/download/getInstallerForGame?os=win&gameProgram=BATTLENET_APP&version=2.26.0.14494

Reply:
From what I can see, we can use https://blizztrack.com/swagger/index.html in an update script, but I don't have time to fiddle with it right now.

Reply:
Ill see what I can do, thank you. Probably will take some time, because of minimal free time